server/api/app.go

```diff
@@ -41,12 +99,16 @@ type Manifest struct {
 
 	// By default invoke "/install", expanding App, AdminAccessToken, and
 	// Config.
-	Install *Call `json:"install,omitempty"`
+	OnInstall *Call `json:"install,omitempty"`
+	OnDelete  *Call `json:"delete,omitempty"`
+	OnStartup *Call `json:"on_startup,omitempty"`
 
 	// By default invoke "/bindings".
 	Bindings *Call `json:"bindings,omitempty"`
 
 	// Deployment manifest for hostable apps will include path->invoke mappings
+	Functions []Function
+	Assets    []Asset
```

Comment:
Like I said in the other comment, we don't really need each (static) asset to be listed individually, do we? They are all just files in a folder.

Reply:
We do. otherwise Proxy plugin will not be able to reach them. Another thing is that **app developer** should not do that. This will be done by the provisioner.